### ISSUE TITLE
Implement bookmarks

### ISSUE THREAD

**Issue by lloydwatkin** [xmpp-ftw/xmpp-ftw#80]
http://xmpp.org/extensions/xep-0048.html

**Comment by santiago26** [comment]
Interesting for me. Additional XEPs should be added based on xmpp-ftw-wtf?

**Comment by lloydwatkin** [comment]
No, that's a generic catch all module for unknown stanzas. If you want to pick this up then I'd suggest something which has been created recently such as https://github.com/xmpp-ftw/xmpp-ftw-register

**Comment by santiago26** [comment]
There are two standards: [ver.2003](http://www.xmpp.org/extensions/attic/xep-0048-1.0.html) - via iq and [ver.2007](http://xmpp.org/extensions/xep-0048.html) - via PubSub. Which should be implemented? As far as I know, the first is common used ( e.g., it used in Psi+).
By the way, how many clients and servers implement 0048 via PubSub(ver 2007), not via iq? Saving bookmarks over PubSub seems more complicated, do we really need it?

Also asking you for an advice, which events have to be implemented? I've read the standard and decided to start work on three events:
- xmpp.bookmarks.get
- xmpp.bookmarks.conference
- xmpp.bookmarks.url

Is it ok?

**Comment by lloydwatkin** [comment]
I'd be tempted to have events:

* xmpp.bookmarks.get
* xmpp.bookmarks.set
* xmpp.bookmarks.push

I wouldn't separate out urls and conferecnes (although I admittedly only had a quick read of the XEP).

The the payload on receiving bookmarks could be something like:

```json
{
  "form": null, /* Potential dataform */
  "items": [
    { "type": "url", "name": "some-name", "url": "http://some.url" },
    { "type": "conference", "name": "some-muc", "autojoin": true, "nick": "its-me!", "jid": "some@chat.room", "password": "areyouserious?!" }
  ]
}
```

**Comment by santiago26** [comment]
.set event  should just form a local object that will later be sent by the .push event. That the idea?
